kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue

```diff
@@ -620,56 +617,58 @@
       channelTitle() {
         return this.channel ? this.channel.name : '';
       },
-      resources() {
-        return this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
-      },
       childrenToDisplay() {
         return this.windowBreakpoint === 2 || this.windowBreakpoint > 4 ? 4 : 3;
       },
       gridType() {
         return this.windowBreakpoint > 4 ? 2 : 1;
       },
-      resourcesDisplayed() {
-        // if no folders are shown at this level, show more resources to fill the space
-        // or if the user has explicitly requested to show more resources
-        if (!this.topics.length || this.showMoreResources) {
-          return this.resources;
-        }
-        return this.resources.slice(0, this.childrenToDisplay);
-      },
-      moreResources() {
-        return this.resourcesDisplayed.length < this.resources.length;
-      },
       topics() {
         return this.contents
           .filter(content => content.kind === ContentNodeKinds.TOPIC)
-          .filter(t => t.children && t.children.results.length)
-          .map(t => {
-            let topicChildren = t.children.results;
+          .filter(t => t.children && t.children.results.length);
+      },
+      contentsForDisplay() {
+        return this.contents
+          .filter(
+            c =>
+              c.kind !== ContentNodeKinds.TOPIC ||
+              ((this.subTopicId ? c.id === this.subTopicId : true) &&
+                c.children &&
+                c.children.results.length),
+          )
+          .reduce((arr, c) => {
+            // Reduce the list to objects representing topics,
+            // and arrays representing runs of resources positioned between them.
+            if (c.kind !== ContentNodeKinds.TOPIC) {
+              const lastEntry = arr.slice(-1)[0];
+              // The final entry is either undefined or not an array
```

Comment:
> The final entry is either undefined or not an array
why?

Reply:
This comment is describing the if statement below, so 'if' this is the case.